### DISCUSSION TITLE
Group LDAP (probably asked a billion times)

### DISCUSSION THREAD

**Discussion by ArnaudTe** [requarks/wiki#6271] [Help / Questions]
Hi,
I have no idea how to setup the LDAP Groupmapping part.
I've read somewhere that you have to create manually the groups but that's the only real information I could find.
What should should be put in the different field if say for example all my groups are in OU=GROUPS,DC=test,DC=DOM ? (and for example one of the groups would be called AccessWikiAdmin)
After that how do I know anything works ? Does the users automically go in the corresponding groups ?
I have tried changing values and activating logs but nothing shows, the only thing I saw was that sometimes I wouldn't be able to log in after some changes were made.

Thanks in advance !
Arnaud

**Comment by jko-nic** [comment]
Hi, I have a new installation (version 2.5.300). I tried altering almost every setting regarding Mapping LDAP Groups, but without any success.

Our LDAP structure is as follows:
- `ou=People, dc=...` node:
  - this holds our users, like `uid=sclaus, ou=People, dc=...`
- `ou=Group, dc=...` node:
  -  there are groups as `ou`, such as `ou=administrators`:
    - every group has an attribute `cn` set as the name of the group, for example `cn=administrators`
    - every group has an attribute `uniqueMember`:
      - this attribute has multiple records of users, who are members of this group, for example `uid=sclaus,ou=People,dc=...` or  `uid=emusk,ou=People,dc...`

Group Search Scope can be `base`, there are no subgroups in `ou=Group,dc=...`. But with other settings I do not know, what should be there. If anyone achieved to make it working, could you please help me? Thanks a lot and have a nice day :)

**Reply by P4mes4n** [reply]
If you are talking about nested groups take a look at [this](https://confluence.atlassian.com/crowdkb/active-directory-user-filter-does-not-search-nested-groups-715130424.html).
We have LDAP Group mappings working just fine.

**Reply by jko-nic** [reply]
No, we do not have nested groups. If your LDAP structure is similar to our (what I posted), could you please share your configuration, what your Group Search Filter, Group DN Property and Group name fields are? Thanks in advance.

**Reply by P4mes4n** [reply]
Forgive formatting, I am on mobile.
First make sure your Groups are named exactly the same in AD and Wikijs. Turn on self registration.
For this assume the domain is "domain.local"

You probably need to adjust these for your own setup, but this works for us.

Group Search Base (OU in which the groups are, that you want to map, in our Case we have Groups 'TeamA' and 'TeamB' in an OU 'Wiki':
'OU=Wiki,OU=Systems,dc=domain,dc=local'

Group Search Filter (this is for nested groups:
'(member:1.2.840.113556.1.4.1941:={{dn}})'

Group search scope:
'sub'

Group DN property:
'dn'

Group name field:
'name'

**Reply by jko-nic** [reply]
Thanks, I deleted Groups Search filter, which appears not to be necessary for NOT nested groups (maybe?), login through LDAP works. Now I must sync group information from LDAP properly, because despite group being created previously in wiki.js), user is still not added to proper group, even if he is part of corresponding LDAP group. Maybe it will be part of [3.0 release](https://js.wiki/feedback/p/group-mapping)?

**Reply by P4mes4n** [reply]
I think you still need a Group Search Filter, but just mapping it on the dn. 
As I said, it already works in the current version, not sure why it isnt for you.
